docs/v1-proposed-api.md

```diff
@@ -0,0 +1,272 @@
+  pullSecret:
+    name: pull
+status:
+  apiURL: https://api.ci-cluster-v4-1.r4f1.s1.devshift.org:6443
+  certificateBundles:
+  - generated: true
+    name: primary-cert-bundle
+  cliImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:528f2ead3d1605bdf818579976d97df5dd86df0a2a5d80df9aa8209c82333a86
+  clusterVersionStatus:
+    [SNIP]
+  conditions:
+  - lastProbeTime: "2019-10-23T08:40:24Z"
+    lastTransitionTime: "2019-10-23T08:40:24Z"
+    message: Control plane certificates are present
+    reason: ControlPlaneCertificatesFound
+  installedTimestamp: "2019-10-23T08:38:03Z"
+  installerImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c9c1e6d1897b74d366d331f5cdc3fa16c5a96abe28ff8238833c09042156e983
+  provision:
+    name: ci-cluster-v4-1-0-ndv7l
+  webConsoleURL: https://console-openshift-console.apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+```
+
+## ClusterDeprovisionRequest
+
+ 1. Rename to ClusterDeprovision, to better match ClusterProvision.
+
+## SyncSet + SelectorSyncSet
+
+ 1. Drop support for ApplyOnce patch mode.
+
+## HiveConfig
+
+ 1. Adjust HiveConfig API to map managed DNS domains to cloud provider they should be used for.
+   * Would allow using Hive to provision to multiple clouds, using that cloud for DNS forwarding, from one Hive.
+   * This is not the expected deployment model in Dedicated (one Hive per cloud)
+   * Convert managedDomains to a list of structs rather than just a string. Struct should include a list of the platforms we will manage DNS for on this sub-domain.
+   * Could have multiple domains on AWS using different certs.
+ 1. Rename ExternalDNS to ManageDNS (section where the credentials are stored, ExternalDNS being phased out)
+
+
+# Migrating to V1
+
+CRD versioning functionality is not available in OpenShift 3.11 where prod/stage run and will continue to do so until private cluster support is ready and we can migrate. We must however move on V1 now.
+
+To avoid having to fork and abandon stage/prod until they can move to 4.3+ we propose the following solution using an aggregated API server.
+
+  1. Add new Hive CRDs, same names as old (aside from those being renamed), but in a new API group.
+  1. Hive controller code all ported to use the new types. No code still references the old.
+  1. Hive operator performs the migration:
+    1. Shut down hive-controllers. We likely will want to coordinate scaledown for all SRE operators as well.
+    1. Copy all types from old to new.
+      * New ClusterDeployment should have no InstallConfig referenced as there is little point in migrating data that is install time only for clusters that are now installed. Some data will be lost. (network config, other things being removed)
+    1. Remove finalizers on old objects to help ensure nothing takes action on them.
+    1. Delete all v1alpha1 custom resources.
+      * Obviously the most risky part of this. Finalizers should be gone and controllers scaled down so no action should be taken.
```

Comment:
Finalizers should not be present at this moment and controllers should have been scaled down so no action would take place.